contrib/pkg/createcluster/create.go

```diff
@@ -263,21 +281,21 @@ func (o *Options) GenerateObjects() ([]runtime.Object, error) {
 	if imageSet != nil {
 		result = append(result, imageSet)
 	}
+
+	// Adding pull secret name to cluster deployment if pull secret exists
+	if pullSecret != nil {
+		cd.Spec.PullSecret = &corev1.LocalObjectReference{
+			Name: fmt.Sprintf("%s-pull-secret", cd.Name),
+		}
+	}
 	result = append(result, cd)
 	if !o.IncludeSecrets {
 		return result, nil
 	}
-	awsCreds, err := o.generateAWSCredsSecret()
-	if err != nil {
-		return nil, err
-	}
-	pullSecret, err := o.generatePullSecret()
-	if err != nil {
-		return nil, err
-	}
-	sshSecret, err := o.generateSSHSecret()
-	if err != nil {
-		return nil, err
+
+	// With introduction of global pull secret, pull secret is optional now
+	if pullSecret == nil {
```

Comment:
Is the ordering of these objects important in some way? If not, then it would be better to do the following.
```
result = append(result, awsCreds, sshSecret)
if pullSecret != nil {
  result = append(result, pullSecret)
}
return result, err
```